eth/beacon/helpers.py

```diff
@@ -1,17 +1,167 @@
 from typing import (
     Iterable,
+    Any,
     List,
+    Tuple,
     TYPE_CHECKING,
 )
 
 from eth_utils import (
     to_tuple,
 )
 
+from eth_typing import (
+    Hash32,
+)
+
+from eth.utils.blake import (
+    blake,
+)
+from eth.beacon.types.shard_and_committee import (
+    ShardAndCommittee,
+)
+
+
 if TYPE_CHECKING:
+    from eth.beacon.config import BeaconConfig  # noqa: F401
+    from eth.beacon.types.active_state import ActiveState  # noqa: F401
+    from eth.beacon.types.attestation_record import AttestationRecord  # noqa: F401
+    from eth.beacon.types.block import Block  # noqa: F401
+    from eth.beacon.types.crystallized_state import CrystallizedState  # noqa: F401
     from eth.beacon.types.validator_record import ValidatorRecord  # noqa: F401
 
 
+#
+# Get block hashes
+#
+def get_block_hash(
+        active_state: 'ActiveState',
+        current_block: 'Block',
+        slot: int,
+        beacon_config: 'BeaconConfig') -> Hash32:
+    """
+    Get the blockhash from active_state.recent_block_hashes by current block slot number.
+    """
+    cycle_length = beacon_config.cycle_length
+
+    sback = current_block.slot_number - cycle_length * 2
+    assert sback <= slot < sback + cycle_length * 2
+    return active_state.recent_block_hashes[slot - sback]
+
+
+@to_tuple
+def get_hashes_from_active_state(active_state: 'ActiveState',
+                                 block: 'Block',
+                                 from_slot: int,
+                                 to_slot: int,
+                                 beacon_config: 'BeaconConfig') -> List[Hash32]:
+    hashes = [
+        get_block_hash(
+            active_state,
+            block,
+            slot,
+            beacon_config,
+        )
+        for slot
+        in range(from_slot, to_slot + 1)
+    ]
+    return hashes
+
+
+@to_tuple
+def get_hashes_to_sign(active_state: 'ActiveState',
+                       block: 'Block',
+                       beacon_config: 'BeaconConfig') -> List[Hash32]:
+    """
+    Given the head block to attest to, collect the list of hashes to be
+    signed in the attestation.
+    """
+    cycle_length = beacon_config.cycle_length
+
+    hashes = get_hashes_from_active_state(
```

Comment:
We can reduce some of the overhead here by changing this function to return `Iterable[Hash32]` and adjusting these lines to be:

```python
yield from get_hashes_from_active_state(...)
yield blake(block.hash)
```

In the current state we have to build the enumerable of hashes twice.